### PR TITLE
feature/SOF-4354 - Add tutorial for training ML workflows

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: lang/en/docs/software-directory/machine-learning/python-ml/components.md [hunk 1]
```diff
@@ -86,7 +89,17 @@ when [ the predict workflow is generated](../../../properties-directory/non-scal
...
+users are expected to modify, to add or remove different machine learning workflow units. In the above diagram,
+we find the following units:
+
+| Order | Unit Name                     | Flavor                                       |
+|-------|-------------------------------|----------------------------------------------|
+| 1     |`Setup Packages and Variables` | `pyml:setup_variables_packages`              |
+| 2     | ` Data Input`                 | `pyml:data_input:read_csv:pandas`            |
+| 3     | `Data Standardize`            | `pyml:pre_processing:standardization:sklearn`|
+| 4     | `Model Train and Predict`     | `pyml:model:multilayer_perceptron:sklearn`   |
+| 5     | `Parity Plot`                 | `pyml:post_processing_parity_plot_matplotlib`|
```
Comment (timurbazhirov) [inline lang/en/docs/software-directory/machine-learning/python-ml/components.md:101]: Shouldn't this be `pyml:post_processing:parity_plot:matplotlib`??

File: lang/en/docs/tutorials/python-ml/train-ml-model.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+![ML Workflow Tab](../../images/tutorials/workflows_tab_with_ml_workflow_and_declare_training_data_circled.png "Workflow Tab")
+
+We can now see the "Declare Training Data" IO unit. Because we named our file in dropbox `data_to_train_with.csv`
+earlier, we do not need to modify this unit. In a scenario where we wanted to use a different file name for our training
+data, the [Workflow Designer](../../workflow-designer/overview.md) could be used to modify the value.
+
+![Training Data IO Unit](../../images/tutorials/training_data_io_unit.png "Training Data IO Unit")
+
+!!!warning "A Word of Caution"
+> The _only_ modifications that should be made in the `Set Up the Job` subworkflow are the filenames in the "Declare
```
Comment (timurbazhirov) [inline lang/en/docs/tutorials/python-ml/train-ml-model.md:82]: Is this the correct way to add an admonition??? https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

File: lang/en/docs/software-directory/machine-learning/python-ml/components.md (hunk 1 above)
Comment (AcylSilane) [reply lang/en/docs/software-directory/machine-learning/python-ml/components.md:101]: Ah, that's a typo in the name of the flavor in the web-app. I'll correct it here and ticket out the fix

File: lang/en/docs/tutorials/python-ml/train-ml-model.md (hunk 2 above)
Comment (AcylSilane) [reply lang/en/docs/tutorials/python-ml/train-ml-model.md]: Fixed in the PR that was just merged into this one

**Comment by AcylSilane** [comment]
Merging per https://exabyte.atlassian.net/browse/SOF-4354?focusedCommentId=41318

**Pull request merged**